### PR TITLE
Fix npm search

### LINKED ISSUE #544: Error with searching for generators using search feature of yo CLI
## Type of issue
Bug

## My environment
* *OS version/details*: `Windows 10 Enterprise N (Azure machine)`
* *Node version:* `6.11.4` 
* *npm version:* `5.5.1` 
* *Version of yo :* `2.0.0` 

## Expected behavior
Correctly finding generators when search feature is executed. 

## Current behavior
Yeoman throws the exception when searching for npm generators is executed.

## Steps to reproduce the behavior
I've just installed Yeoman. I run `yo` command. Then from the menu I select `Install a generator`. Then Yeoman asks `Search npm for generators`. I tried a few (like `angular` or `jhipster`). Then I get the error message (see below).

## Command line output

> PS C:\Users\userName\Desktop\angularJS unit tests> yo
> ? 'Allo userName! What would you like to do? Install a generator
> ? Search npm for generators: angular
> 
>      _-----_     ╭───────────────────────╮
>     |       |    │      Bye from us!     │
>     |--(o)--|    │       Chat soon.      │
>    `---------´   │      Yeoman team      │
>     ( _´U`_ )    │    http://yeoman.io   │
>     /___A___\   /╰───────────────────────╯
>      |  ~  |
>    __'.___.'__
>  ´   `  |° ´ Y `
> 
> ParseError: Unexpected token '<' at 2:1 in https://registry.npmjs.org/-/_view/byKeyword?startkey=[%22yeoman-generator%22]&endkey=[%22yeoman-generat
> or%22,%7B%7D]&group_level=3
> <?xml version="1.0" encoding="utf-8"?>
> ^
>     at C:\Users\userName\AppData\Roaming\npm\node_modules\yo\node_modules\npm-keyword\node_modules\got\index.js:120:14
>     at BufferStream.<anonymous> (C:\Users\userName\AppData\Roaming\npm\node_modules\yo\node_modules\read-all-stream\index.js:64:3)
>     at emitNone (events.js:91:20)
>     at BufferStream.emit (events.js:185:7)
>     at finishMaybe (C:\Users\userName\AppData\Roaming\npm\node_modules\yo\node_modules\readable-stream\lib\_stream_writable.js:607:14)
>     at afterWrite (C:\Users\userName\AppData\Roaming\npm\node_modules\yo\node_modules\readable-stream\lib\_stream_writable.js:470:3)
>     at _combinedTickCallback (internal/process/next_tick.js:86:20)
>     at process._tickCallback (internal/process/next_tick.js:104:9)
> PS C:\Users\userName\Desktop\angularJS unit tests>

### PR DESCRIPTION
**Pull request by mischah**
Closes #544

## Purpose of this pull request? 

- [ ] Documentation update
- [x] Bug fix 
- [ ] Enhancement
- [ ] Other, please explain:

## What changes did you make?

Updated npm-keyword to use the new npm API to search for Yeoman generators.

Since the new API restricts the number of results we need to pass the combination of `yeoman-generator` + `searchterm` to npm-keyword.

## Is there anything you'd like reviewers to focus on?

I wonder if it’s still necessary to `memoize` the result since the result of this request is now going to change with every different searchterm  and the result is limited to 250 generators:
https://github.com/yeoman/yo/blob/a4b7be3eeb44cdc62b9f018039482ba189a94875/lib/routes/install.js#L44

### CODE REVIEW

File: lib/routes/install.js [hunk 1]
```diff
@@ -41,13 +41,13 @@ module.exports = app => {
 };
 
 const generatorMatchTerm = (generator, term) => `${generator.name} ${generator.description}`.indexOf(term) !== -1;
-const getAllGenerators = _.memoize(() => npmKeyword('yeoman-generator'));
+const getGenerators = _.memoize(term => npmKeyword(['yeoman-generator', term]));
```
Comment (SBoudrias) [inline lib/routes/install.js]: Should we still re-filter by ourselves later on then?

`generatorMatchTerm(generator, term);`

Comment (mischah) [reply lib/routes/install.js]: Uhm. No. Guess it was a little late last night 🙈 
Going to remove that.

**Comment by mischah** [comment]
I removed the filtering. But those failing tests are driving me nuts 🙀 

To be honest I just don’t get what’s happening there in depth. Tried to fix em really hard. But Cant get it right.

Would love to get some help 🆘

**Comment by SBoudrias** [comment]
@mischah my guess is that `nock(registryUrl).get('/-/v1/search')` doesn't actually catch the service call.

**Pull request closed without merging**